### DISCUSSION TITLE
Logpoints!

### DISCUSSION THREAD

**Discussion by fredrik-m-j** [prb28/vscode-amiga-assembly#249] [Show and tell]
First of all - the Logpoint feature is awsome!

I have been struggling in the past with debugging code which uses series of random data added to a queue for later processing. Logpoints has been very helpful here as I can just grab the random data and generate asm from it, like so:

Logpoint expression (d0 and d1 contains random data):
move.b #{d0.b},(a0)+
move.b #$cd,(a0)+
move.w #{d1.w},(a0)+
Logpoints will accept and output newlines too 👍 

Example output in debug console:
move.b #0x33,(a0)+
move.b #$cd,(a0)+
move.w #0x24d,(a0)+
move.b...
etc.

I copy the output and paste it into my test harness (that populates my queue) and now I can replay and debug that particular random series in detail.

There is one additional step - I have to replace the hex prefix 0x with $. Perhaps there is a way to do it in the formatting of the logpoint?

One thing I could propose as an improvement is make the logpoint more visually different from the breakpoint. I accidentally deleted a logpoint because I thought it was an irrelevant breakpoint. Perhaps using a completely different color?

**Comment by prb28** [comment]
The logpoint color is set by vscode itself, maybe you can find some extension or setting that can set it's color.

**Reply by fredrik-m-j** [reply]
Ok, I didn't realize its color is set by VS Code. Good to know. Anyway, it's not a problem - just a very very small inconvenience. :-)
